File: Data Structures & Algorithms/house-robber-ii/submission-3.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        if(nums.size() < 2)
            return nums[0];
        

        return max(helper(1, nums.size(), nums), helper(0, nums.size() - 1, nums));
    }

    int helper(int start, int end, vector<int>& nums)
    {
        int prev1 = 0;
        int prev2 = 0;

        for(int i = start; i < end; i++ )
        {

            int cur = max(prev1, prev2 + nums[i]);
            prev2 = prev1;
            prev1 = cur;
        }

        return prev1;
    }
};

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-3.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        
        cur_pos = 1

        for i in range(1, len(nums)):
            if nums[i] != nums[i - 1]:
                nums[cur_pos] = nums[i]
                cur_pos += 1
        
        return cur_pos